dask/dataframe/io/parquet/arrow.py

```diff
@@ -103,33 +116,49 @@ def _determine_dataset_parts(fs, paths, gather_statistics, filters, dataset_kwar
     """
     parts = []
     if len(paths) > 1:
-        if gather_statistics is not False:
-            # This scans all the files
-            dataset = pq.ParquetDataset(
-                paths, filesystem=fs, filters=filters, **dataset_kwargs
-            )
-        else:
-            base, fns = _analyze_paths(paths, fs)
-            if "_metadata" in fns:
-                # We have a _metadata file, lets use it
+        base, fns = _analyze_paths(paths, fs)
+        if "_metadata" in fns:
+            # We have a _metadata file
+            # PyArrow cannot handle "_metadata"
+            # when `paths` is a list.
+            paths.remove(base + fs.sep + "_metadata")
+            fns.remove("_metadata")
+            if gather_statistics is not False:
+                # If we are allowed to gather statistics,
+                # lets use "_metadata" instead of opening
+                # every file. Note that we don't need to check if
+                # the dataset is flat here, because PyArrow cannot
+                # properly handle partitioning in this case anyway.
```

Comment:
Can you explain a bit more what pyarrow is not handling? (also to understand if something needs to be fixed on our side)

Reply:
PyArrow correctly handles the existence of a "_metadata" file when the user passes a directory to `ParquetDataset`.  That is, the "_metadata" file will be used to populate the `metadata` attribute, and the list of `ParquetDatasetPiece` objects will not include the "_metadata" file (because it does not include any data).

However, when the user passes in a list of files, with one being "_metadata", the `metadata` attribute is **not** populated, and the "_metadata" path is included as one of the `ParquetDatasetPiece` objects instead (which leads to a n `ArrowIOError` during the read of that piece).  For example:

```python
import dask
import glob
import pyarrow.parquet as pa

path = <path>
dask.datasets.timeseries(freq="1D").to_parquet(path)

paths = glob.glob(path + "/*.parquet")
paths.append(path + "/_metadata")
dataset = pa.ParquetDataset(paths)

# Last piece will be the _metadata file,
# which leads to an ArrowIOError during read:
dataset.pieces[-1].read()
```

Reply:
OK, thanks for the explanation! I opened https://issues.apache.org/jira/browse/ARROW-8446 to keep track of this on the PyArrow side